tests/ui/layout/reprc-power-alignment.rs

```diff
@@ -0,0 +1,145 @@
+//@ check-pass
+//@ revisions: aix
+//@[aix] compile-flags: --target powerpc64-ibm-aix
+//@[aix] needs-llvm-components: powerpc
+
+#![warn(uses_power_alignment)]
+
+#[repr(C)]
+pub struct Floats {
+    a: f64,
+    b: u8,
+    c: f64, //~ WARNING repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
+    d: f32,
+}
+
+pub struct Floats2 {
+    a: f64,
+    b: u32,
+    c: f64,
+}
+
+#[repr(C)]
+pub struct Floats3 {
+    a: f32,
+    b: f32,
+    c: i64,
+}
+
+#[repr(C)]
+pub struct Floats4 {
+    a: u64,
+    b: u32,
+    c: f32,
+}
+
+#[repr(C)]
+pub struct Floats5 {
+    a: f32,
+    b: f64, //~ WARNING repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
+    c: f32,
+}
+
+#[repr(C)]
+pub struct FloatAgg1 {
+    x: Floats,
+    y: f64, //~ WARNING repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
+}
+
+#[repr(C)]
+pub struct FloatAgg2 {
+    x: i64,
+    y: Floats, //~ WARNING repr(C) does not follow the power alignment rule. This may affect platform C ABI compatibility for this type
```

Comment:
why would this be linted on? the first field isn't a float. ideally it should lint only on the struct that actually causes the problem (has a float first and then includes another float-containing type), not the structs that are "innocent bystanders"

Reply:
note that if you don't care to fix this, that's fine, fortunately this lint doesn't affect other targets...

Reply:
Yeah, you're right - thanks for bringing this up. Technically this struct would be a false positive in this case from my implementation and in the ideal case, I agree that we don't need to flag this struct.

That being said, I also don't plan to fix this at this point in time and have anticipated that although some false positives may be detected, detecting a majority/all of the problematic cases is the priority.

Reply:
That's fine then. A false positive in a lint is undesirable but I agree a false negative would be much worse.